server/src/main/java/io/spine/server/event/model/EventAcceptor.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2018, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.event.model;
+
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.Message;
+import io.spine.core.CommandClass;
+import io.spine.core.CommandContext;
+import io.spine.core.DispatchedCommand;
+import io.spine.core.EventClass;
+import io.spine.core.EventContext;
+import io.spine.core.EventEnvelope;
+import io.spine.server.model.HandlerKey;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.ImmutableList.copyOf;
+import static com.google.common.collect.ImmutableList.of;
+import static io.spine.protobuf.AnyPacker.unpack;
+import static io.spine.util.Exceptions.newIllegalStateException;
+
+/**
+ * The strategy of event accepting method invocation.
+ *
+ * <p>An event accepting method is an event {@linkplain io.spine.core.Subscribe subscriber} or
+ * an event {@link io.spine.server.event.React reactor}.
+ *
+ * @author Dmytro Dashenkov
+ */
+enum EventAcceptor {
+
+    /**
+     * Method accepting only the event message.
+     */
+    MESSAGE(of(Message.class), false) {
+        @Override
+        List<?> arguments(EventEnvelope envelope) {
+            Message message = envelope.getMessage();
+            return of(message);
+        }
+    },
+
+    /**
+     * Method accepting the event message and the {@link EventContext}.
+     */
+    MESSAGE_EVENT_CXT(of(Message.class, EventContext.class), false) {
```

Comment:
`CXT` seems to be an airport. `CTX` is also an airport, but also stands for "context".